idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1462,3 +1462,91 @@ struct HistoryBranch{
   20: optional string branchID
   30: optional list<HistoryBranchRange>  ancestors
 }
+
+enum BatchOperationType{
+    Terminate
+    RequestCancel
+    Signal
+}
+
+struct BatchSignalRequest{
+    10: optional string signalName
+    20: optional binary signalData
+}
+
+struct BatchTerminateRequest{
+    10: optional bool applyOnChildren
```

Comment:
Let's don't put it to not deprecate as soon as parent policy is implemented.

Reply:
After the meeting I think it a little more that we still need it. 
ParentClosePolicy won't help because a lot of exsiting domains will be using abandon as default or setting abandon and change mind later because. 
We thought replacement is after we support RootWorkflowID as search attributes, then customer can use batch with querying RootWorkflowID. But it is not correct:  a customer can have multiple trees and the RootWorkflowIDs are from another query. In that case this is the only way a customer can kill the trees because our ES query doesn't support subQuery(or JOIN). 
And and it is always easier for customers to understand with this parameter rather than using RootWorkflowID.